### ISSUE TITLE
Wrong gradient when using backslash

### ISSUE THREAD

**Issue by baggepinnen** [FluxML/Zygote.jl#222]
The following example produces different results when using Zygote and ForwardDiff
```julia
using ForwardDiff, Zygote
r(a) = norm(a[1:end-1,:]\a[2:end,:])
Zygote.refresh()
a = randn(33,3)
norm(Zygote.gradient(r,a)[1] - ForwardDiff.gradient(r,a)) # ~= 0.2
```

**Comment by willtebbutt** [comment]
Yup, this is definitely a thing. Working on a fix at the minute. In short, the current implementation assumes that matrix on the lhs is square, and that therefore the regular inverse is used, rather than the pseudo-inverse.

**Comment by MikeInnes** [comment]
Is this fixed by #223?

**Comment by DoktorMike** [comment]
This is fixed for me on master.

```julia
using ForwardDiff, Zygote, LinearAlgebra
r(a) = norm(a[1:end-1,:]\a[2:end,:])
Zygote.refresh()
a = randn(33,3)
norm(Zygote.gradient(r,a)[1] - ForwardDiff.gradient(r,a)) # ~= 2.0503073816403344e-16
```
